### DISCUSSION TITLE
[Mining Issue] Very low lucky vaulue

### DISCUSSION THREAD

**Discussion by scvpark** [filecoin-project/lotus#6002] [Storage Provider] [answered]
> Note: For security-related bugs/issues, please follow the [security policy](https://github.com/filecoin-project/lotus/security/policy).

Please provide all the information requested here to help us troubleshoot "mining/WinningPoSt failed" issues.
If the information requested is missing, you may be asked you to provide it.

**Describe the problem**
I mined for more than 3 months, but I got only 7 blocks. It's a fairly low number compared to other minor numbers. I think this is not unlucky, but there are other problems. 
And I have an ID with similar power in other regions, and I get blocks every day from other IDs.

**Version**
from March to April : 1.5.0 

2021-04-09 14:00 KST updated miner version.
lotus version 1.6.0+mainnet+git.3fc23a785

**Setup**
No firewall.  
Only NAT 
miner id : f0110768
GPU : RTX 3090
CPU : AMD Ryzen Threadripper 3970X 32-Core Processor
variable
source "$HOME/.cargo/env"
export PATH=$PATH:$HOME/.cargo/bin:/usr/local/go/bin
export LOTUS_MINER_PATH=/usr/seal/lotusminer
export LOTUS_PATH="/usr/seal/.lotus"
export CGO_CFLAGS_ALLOW="-D__BLST_PORTABLE__"
export CGO_CFLAGS="-D__BLST_PORTABLE__"
export RUSTFLAGS="-C target-cpu=native -g"
export FFI_BUILD_FROM_SOURCE=1
export BELLMAN_CUSTOM_GPU="Geforce RTX 3090:10496"
export FIL_PROOFS_MAXIMIZE_CACHING=1
export FIL_PROOFS_USE_GPU_COLUMN_BUILDER=1
export FIL_PROOFS_USE_GPU_TREE_BUILDER=1
export FIL_PROOFS_USE_MULTICORE_SDR=1
export LOTUS_BACKUP_BASE_PATH=/root/lotus_backups/
export FIL_PROOFS_PARAMETER_CACHE=/var/tmp/parameters
export FIL_PROOFS_PARENT_CACHE=/var/tmp/parents
export GOLOG_FILE="/root/miner.log"
export GOLOG_LOG_FMT="json"

when i start lotus daemon : 
nohup lotus daemon > /root/lotus.log 2>&1 &
when start lotus-miner  : 
nohup lotus-miner run > ~/lotus-miner.log 2>&1 &

**Lotus daemon and miner logs**
After getting the first block on January 21, 2021, we have continuously mined it, but so far we have 7 blocks.
I can see the mined block log on April 6th, but I didn't actually get it.

![image](https://user-images.githubusercontent.com/82202610/114143064-a7675d00-994e-11eb-819d-7f2d79af7fee.png)
![image](https://user-images.githubusercontent.com/82202610/114143128-b8b06980-994e-11eb-8e53-ccf60a1b73cd.png)

** Code modifications **
Nothing has been modified.

**Comment by jennijuju** [comment]
Converting to a discussion for community discussion - please open a new tickets when there is a more defined issue or request.

**Comment by dd45e640b42e6da7da96faee3996ef7c** [comment, marked as answer]
did the issue resolve? i saw you won quite a few since this post 8 days ago.

did you scan your logs for missed blocks - either computed too slow or not included in the chain?

**Reply by scvpark** [reply]
I scanned the log, but there was no slow compute or not included in the chain.
My power is 320Tb now, but the block hasn't come out for 2 days.
My win count value is abnormally low compared to other minor and win count with similar power.
I really don't know what the problem is..

**Reply by dd45e640b42e6da7da96faee3996ef7c** [reply]
until the mid of march i assume your block win rate was around 0.25 / day 
(75TB). then the big sealing started. now it should be 1/day. just looking at you pledge amounts here - do not pin me these numbers please...

it is still very low, you are very right about that.

the only 3 things that i can think of are:

- very very very bad luck, i doubt it honestly
- missed blocks due to too slow calculation
- network issues

can you run

cat log.file | grep "mined" | wc -l

on your log file. i hope you have  full history log file for the miner

**Reply by scvpark** [reply]
My minor log has been recorded since April 1st.
cat miner.log | grep -i mined | wc -l 
7

I checked my external internet upload and download speed.

Retrieving speedtest.net server list...
Selecting best server based on ping...
Hosted by kdatacenter.com (Seoul) [270.72 km]: 69.96 ms
Testing download speed................................................................................
Download: 229.51 Mbit/s
Testing upload speed......................................................................................................
Upload: 202.21 Mbit/s
root@filecoin-main:~#

**Comment by dd45e640b42e6da7da96faee3996ef7c** [comment]
did you ever run this

https://docs.filecoin.io/mine/lotus/miner-setup/#publishing-the-miner-addresses

?

**Reply by scvpark** [reply]
yes i did run it.
lotus-miner actor set-addrs /ip4/211.223.2.211/tcp/24001

root@filecoin-main:~# lotus-miner net listen
/ip4/211.223.2.211/tcp/24001/p2p/12D3KooWHU6NdZNpMBZjZ7U5Jz2sBYMR5b7zyXmYNiYBCFRRrJLR

root@filecoin-main:~# lotus-miner net reachability
AutoNAT status:  Public
Public address:  /ip4/211.223.2.211/tcp/41783

root@filecoin-main:~# lotus-miner net peers | wc -l
123

**Reply by scvpark** [reply]
Today i get 2 blocks.. 
I will wait a few more days..

**Reply by scvpark** [reply]
I'm not getting a block again after the 19th. 
I often see the following log in my miner log, is it related to this?
{"level":"info","ts":"2021-04-21T15:09:37.882+0900","logger":"swarm2","caller":"go-libp2p-swarm@v0.3.1/swarm_dial.go:533","msg":"got error on dial: dial tcp6 [::1]:31637: connect: connection refused"}
{"level":"info","ts":"2021-04-21T15:09:37.882+0900","logger":"swarm2","caller":"go-libp2p-swarm@v0.3.1/swarm_dial.go:502","msg":"got error on dial: dial tcp4 127.0.0.1:21825: connect: connection refused"}
{"level":"info","ts":"2021-04-21T15:09:37.882+0900","logger":"swarm2","caller":"go-libp2p-swarm@v0.3.1/swarm_dial.go:533","msg":"got error on dial: dial tcp4 127.0.0.1:2347: connect: connection refused"}
{"level":"info","ts":"2021-04-21T15:09:37.882+0900","logger":"swarm2","caller":"go-libp2p-swarm@v0.3.1/swarm_dial.go:502","msg":"got error on dial: dial tcp6 [::1]:2347: connect: connection refused"}
{"level":"info","ts":"2021-04-21T15:09:37.889+0900","logger":"swarm2","caller":"go-libp2p-swarm@v0.3.1/swarm_dial.go:533","msg":"got error on dial: dial tcp6 [::1]:35443: connect: connection refused"}
{"level":"info","ts":"2021-04-21T15:09:37.889+0900","logger":"swarm2","caller":"go-libp2p-swarm@v0.3.1/swarm_dial.go:502","msg":"got error on dial: dial tcp4 127.0.0.1:38169: connect: connection refused"}
{"level":"info","ts":"2021-04-21T15:09:37.967+0900","logger":"swarm2","caller":"go-libp2p-swarm@v0.3.1/swarm_dial.go:533","msg":"got error on dial: dial tcp6 [::1]:45369: connect: connection refused"}
{"level":"info","ts":"2021-04-21T15:09:37.967+0900","logger":"swarm2","caller":"go-libp2p-swarm@v0.3.1/swarm_dial.go:502","msg":"got error on dial: dial tcp4 127.0.0.1:40861: connect: connection refused"}
{"level":"info","ts":"2021-04-21T15:09:37.984+0900","logger":"swarm2","caller":"go-libp2p-swarm@v0.3.1/swarm_dial.go:533","msg":"got error on dial: dial tcp4 119.28.69.134:26913: connect: connection refused"}
{"level":"info","ts":"2021-04-21T15:09:38.017+0900","logger":"swarm2","caller":"go-libp2p-swarm@v0.3.1/swarm_dial.go:533","msg":"got error on dial: dial tcp4 127.0.0.1:46613: connect: connection refused"}
{"level":"info","ts":"2021-04-21T15:09:38.017+0900","logger":"swarm2","caller":"go-libp2p-swarm@v0.3.1/swarm_dial.go:533","msg":"got error on dial: dial tcp6 [::1]:36445: connect: connection refused"}
{"level":"info","ts":"2021-04-21T15:09:38.062+0900","logger":"swarm2","caller":"go-libp2p-swarm@v0.3.1/swarm_dial.go:533","msg":"got error on dial: dial tcp6 [::1]:42409: connect: connection refused"}
{"level":"info","ts":"2021-04-21T15:09:38.062+0900","logger":"swarm2","caller":"go-libp2p-swarm@v0.3.1/swarm_dial.go:502","msg":"got error on dial: dial tcp4 127.0.0.1:34077: connect: connection refused"}
{"level":"info","ts":"2021-04-21T15:09:38.195+0900","logger":"swarm2","caller":"go-libp2p-swarm@v0.3.1/swarm_dial.go:533","msg":"got error on dial: dial tcp6 [::1]:35265: connect: connection refused"}
{"level":"info","ts":"2021-04-21T15:09:38.195+0900","logger":"swarm2","caller":"go-libp2p-swarm@v0.3.1/swarm_dial.go:502","msg":"got error on dial: dial tcp4 127.0.0.1:35311: connect: connection refused"}
{"level":"info","ts":"2021-04-21T15:09:38.227+0900","logger":"swarm2","caller":"go-libp2p-swarm@v0.3.1/swarm_dial.go:533","msg":"got error on dial: dial tcp4 127.0.0.1:1347: connect: connection refused"}
{"level":"info","ts":"2021-04-21T15:09:38.343+0900","logger":"swarm2","caller":"go-libp2p-swarm@v0.3.1/swarm_dial.go:533","msg":"got error on dial: dial tcp4 127.0.0.1:12340: connect: connection refused"}
{"level":"info","ts":"2021-04-21T15:09:38.343+0900","logger":"swarm2","caller":"go-libp2p-swarm@v0.3.1/swarm_dial.go:502","msg":"got error on dial: dial tcp6 [::1]:12340: connect: connection refused"}
{"level":"info","ts":"2021-04-21T15:09:38.366+0900","logger":"swarm2","caller":"go-libp2p-swarm@v0.3.1/swarm_dial.go:533","msg":"got error on dial: dial tcp6 [::1]:12340: connect: connection refused"}
{"level":"info","ts":"2021-04-21T15:09:38.366+0900","logger":"swarm2","caller":"go-libp2p-swarm@v0.3.1/swarm_dial.go:502","msg":"got error on dial: dial tcp4 127.0.0.1:12340: connect: connection refused"}
{"level":"info","ts":"2021-04-21T15:09:38.379+0900","logger":"swarm2","caller":"go-libp2p-swarm@v0.3.1/swarm_dial.go:533","msg":"got error on dial: dial tcp4 127.0.0.1:1347: connect: connection refused"}
{"level":"info","ts":"2021-04-21T15:09:38.416+0900","logger":"swarm2","caller":"go-libp2p-swarm@v0.3.1/swarm_dial.go:533","msg":"got error on dial: dial tcp4 36.189.234.218:62866: connect: connection refused"}
{"level":"info","ts":"2021-04-21T15:09:38.533+0900","logger":"swarm2","caller":"go-libp2p-swarm@v0.3.1/swarm_dial.go:533","msg":"got error on dial: dial tcp6 [::1]:30669: connect: connection refused"}
{"level":"info","ts":"2021-04-21T15:09:38.533+0900","logger":"swarm2","caller":"go-libp2p-swarm@v0.3.1/swarm_dial.go:502","msg":"got error on dial: dial tcp4 127.0.0.1:32433: connect: connection refused"}
{"level":"info","ts":"2021-04-21T15:09:38.584+0900","logger":"swarm2","caller":"go-libp2p-swarm@v0.3.1/swarm_dial.go:533","msg":"got error on dial: dial tcp4 36.189.234.163:49998: connect: connection refused"}
{"level":"info","ts":"2021-04-21T15:09:38.654+0900","logger":"swarm2","caller":"go-libp2p-swarm@v0.3.1/swarm_dial.go:533","msg":"got error on dial: dial tcp4 127.0.0.1:50888: connect: connection refused"}
{"level":"info","ts":"2021-04-21T15:09:38.655+0900","logger":"swarm2","caller":"go-libp2p-swarm@v0.3.1/swarm_dial.go:533","msg":"got error on dial: dial tcp4 127.0.0.1:42540: connect: connection refused"}
{"level":"info","ts":"2021-04-21T15:09:38.655+0900","logger":"swarm2","caller":"go-libp2p-swarm@v0.3.1/swarm_dial.go:533","msg":"got error on dial: dial tcp6 [::1]:42540: connect: connection refused"}
{"level":"info","ts":"2021-04-21T15:09:38.703+0900","logger":"swarm2","caller":"go-libp2p-swarm@v0.3.1/swarm_dial.go:533","msg":"got error on dial: dial tcp6 [::1]:46043: connect: connection refused"}
{"level":"info","ts":"2021-04-21T15:09:38.703+0900","logger":"swarm2","caller":"go-libp2p-swarm@v0.3.1/swarm_dial.go:502","msg":"got error on dial: dial tcp4 127.0.0.1:36245: connect: connection refused"}
{"level":"info","ts":"2021-04-21T15:09:38.901+0900","logger":"swarm2","caller":"go-libp2p-swarm@v0.3.1/swarm_dial.go:533","msg":"got error on dial: dial tcp4 127.0.0.1:12340: connect: connection refused"}
{"level":"info","ts":"2021-04-21T15:09:38.901+0900","logger":"swarm2","caller":"go-libp2p-swarm@v0.3.1/swarm_dial.go:502","msg":"got error on dial: dial tcp6 [::1]:12340: connect: connection refused"}
{"level":"info","ts":"2021-04-21T15:09:38.938+0900","logger":"swarm2","caller":"go-libp2p-swarm@v0.3.1/swarm_dial.go:533","msg":"got error on dial: dial tcp4 127.0.0.1:5678: connect: connection refused"}
{"level":"info","ts":"2021-04-21T15:09:38.982+0900","logger":"swarm2","caller":"go-libp2p-swarm@v0.3.1/swarm_dial.go:533","msg":"got error on dial: dial tcp4 127.0.0.1:5678: connect: connection refused"}
{"level":"info","ts":"2021-04-21T15:09:38.982+0900","logger":"swarm2","caller":"go-libp2p-swarm@v0.3.1/swarm_dial.go:502","msg":"got error on dial: dial tcp6 [::1]:5678: connect: connection refused"}
{"level":"info","ts":"2021-04-21T15:09:39.034+0900","logger":"swarm2","caller":"go-libp2p-swarm@v0.3.1/swarm_dial.go:533","msg":"got error on dial: dial tcp4 127.0.0.1:41933: connect: connection refused"}
{"level":"info","ts":"2021-04-21T15:09:39.034+0900","logger":"swarm2","caller":"go-libp2p-swarm@v0.3.1/swarm_dial.go:502","msg":"got error on dial: dial tcp6 [::1]:39021: connect: connection refused"}
{"level":"info","ts":"2021-04-21T15:09:39.122+0900","logger":"swarm2","caller":"go-libp2p-swarm@v0.3.1/swarm_dial.go:533","msg":"got error on dial: dial tcp4 126.12.71.119:59374: connect: connection refused"}
{"level":"info","ts":"2021-04-21T15:09:39.138+0900","logger":"swarm2","caller":"go-libp2p-swarm@v0.3.1/swarm_dial.go:533","msg":"got error on dial: dial tcp6 [::1]:34149: connect: connection refused"}
{"level":"info","ts":"2021-04-21T15:09:39.138+0900","logger":"swarm2","caller":"go-libp2p-swarm@v0.3.1/swarm_dial.go:502","msg":"got error on dial: dial tcp4 127.0.0.1:33223: connect: connection refused"}
{"level":"info","ts":"2021-04-21T15:09:39.266+0900","logger":"swarm2","caller":"go-libp2p-swarm@v0.3.1/swarm_dial.go:533","msg":"got error on dial: dial tcp4 127.0.0.1:42859: connect: connection refused"}
{"level":"info","ts":"2021-04-21T15:09:39.266+0900","logger":"swarm2","caller":"go-libp2p-swarm@v0.3.1/swarm_dial.go:502","msg":"got error on dial: dial tcp6 [::1]:43579: connect: connection refused"}
{"level":"info","ts":"2021-04-21T15:09:39.267+0900","logger":"swarm2","caller":"go-libp2p-swarm@v0.3.1/swarm_dial.go:533","msg":"got error on dial: dial tcp4 127.0.0.1:23753: connect: connection refused"}
{"level":"info","ts":"2021-04-21T15:09:39.267+0900","logger":"swarm2","caller":"go-libp2p-swarm@v0.3.1/swarm_dial.go:502","msg":"got error on dial: dial tcp6 [::1]:18057: connect: connection refused"}

**Reply by stuberman** [reply]
No, those are loopback and null addresses which should not affect you.

**Reply by scvpark** [reply]
Oh Thank you for reply.   I don't have to worry about the log anymore.

**Comment by ypool-xiaoyaozi** [comment]
must do ✅ Do you want to run actor? It's necessary to open storage and order receiving. It doesn't seem to be necessary
